src/App/src/Repository/OAuthClientRepository.php

```diff
@@ -52,7 +52,7 @@ public function getClientEntity($clientIdentifier): ?ClientEntityInterface
     public function validateClient($clientIdentifier, $clientSecret, $grantType): bool
     {
         $client = $this->getClientEntity($clientIdentifier);
-        if (! $client instanceof ClientEntityInterface) {
+        if (! $client instanceof OAuthClient) {
```

Comment:
Why switch to check concrete implementation instead of interface?

Reply:
Because the `ClientEntityInterface` doesn't implement the `getSecret()` method, only the concrete implementation does.